designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPassword.kt

```diff
@@ -0,0 +1,140 @@
+                inputTextValue = TextFieldValue(uiModel.inputTextFieldValue?.text ?: "", TextRange(uiModel.inputTextFieldValue?.text?.length ?: 0)),
+                isSingleLine = true,
+                onInputChanged = { newText ->
+
+                    uiModel.onValueChanged?.invoke(newText)
+                },
+                enabled = uiModel.state != InputShellState.DISABLED,
+                state = uiModel.state,
+                keyboardOptions = KeyboardOptions(imeAction = uiModel.imeAction, keyboardType = KeyboardType.Text),
+                visualTransformation = if (showPassword.value) VisualTransformation.None else PasswordVisualTransformation(),
+                onNextClicked = {
+                    if (uiModel.onNextClicked != null) {
+                        uiModel.onNextClicked.invoke()
+                    } else {
+                        focusManager.moveFocus(FocusDirection.Down)
+                    }
+                },
+            )
+        },
+        primaryButton = {
+            if (!uiModel.inputTextFieldValue?.text.isNullOrBlank() && uiModel.state != InputShellState.DISABLED) {
+                IconButton(
+                    modifier = Modifier.testTag("INPUT_PASSWORD_RESET_BUTTON").padding(Spacing.Spacing0),
+                    icon = {
+                        Icon(
+                            imageVector = Icons.Outlined.Cancel,
+                            contentDescription = "Icon Button",
+                        )
+                    },
+                    onClick = {
+                        uiModel.onValueChanged?.invoke(TextFieldValue())
+                        focusRequester.requestFocus()
+                    },
+                )
+            }
+        },
+        secondaryButton = {
+            SquareIconButton(
+                modifier = Modifier.testTag("INPUT_PASSWORD_ACTION_BUTTON")
+                    .focusable(),
+                icon = {
+                    Icon(
+                        imageVector = if (showPassword.value) Icons.Outlined.Visibility else Icons.Outlined.VisibilityOff,
+                        contentDescription = null,
+                    )
+                },
+                onClick = {
+                    showPassword.value = !showPassword.value
+                },
+                enabled = uiModel.state != InputShellState.DISABLED,
+            )
+        },
+        supportingText =
+        {
+            supportingTextList.forEach { item ->
+                SupportingText(
+                    item.text,
+                    item.state,
+                    modifier = Modifier.testTag("INPUT_PASSWORD_SUPPORTING_TEXT"),
+                )
+            }
+        },
+        legend = {
+            uiModel.legendData?.let {
+                Legend(uiModel.legendData, Modifier.testTag("INPUT_PASSWORD_LEGEND"))
+            }
+        },
+        inputStyle = uiModel.inputStyle,
+    )
+}
+
+data class InputPasswordModel(
```

Comment:
Aren't InputPasswordModel and InputUserModel pretty similar? do you think it make sense to think on a common UI model or at least a whay we can all of them in a common list, I'm thinking if we want to have a list of elements (final ui model just to render) the list needs to contains different clases (InputPasswordModel, inputUserModel, InputDateModel...) so we need to map in the UI, but if there is a common class (like the same ui model or sealed models) we can map the list in the presentation layer.
What do you think?

Reply:
updated PR